src/cljs/main/broadfcui/main.cljs

```diff
@@ -120,68 +120,106 @@
                   "http://status.firecloud.org/"]
                  " for more information."])}))
 
-(defn status-alert [title message link]
-  [:div {:style {:borderBottom "1px solid" :borderBottomColor (:line-default style/colors)
-                 :backgroundColor (:exception-state style/colors) :padding "1rem"}}
-   [:div {:style {:display "flex" :background (:exception-state style/colors) :color "#fff"
-                  :alignItems "center" :marginBottom "0.5rem"}}
-    [icons/ExceptionIcon {:size 18}]
-    [:span {:style {:marginLeft "0.5rem" :fontWeight 600
-                    :verticalAlign "middle"}}
-     (if title title "Service Alert")]]
-   [:div {:style {:color "#fff" :fontSize "90%"}}
-    (str message " ")
-     (when link [:a {:href (str link) :target "_blank" :style {:color "#fff"}} "Read more..."])]])
-
 (defn status-alert-interval [attempt]
   (cond
     (= attempt 0) (config/status-alerts-refresh)
     (> attempt (config/max-retry-attempts)) (config/status-alerts-refresh)
     :else (utils/get-exponential-backoff-interval attempt)))
 
-(react/defc StatusAlertContainer
+(react/defc Banner
+  {:get-initial-state
+   (fn []
+     {:showing-more? false})
+   :render
+   (fn [{:keys [props state]}]
+     [:div {}
+      (let [{:keys [background-color text-color title message link more-content]} props]
+        (utils/cljslog more-content)
+        [:div {:style {:borderBottom "1px solid" :borderBottomColor (:line-default style/colors) :color text-color
+                       :backgroundColor background-color :padding "1rem"}}
+         [:div {:style {:display "flex" :alignItems "center" :marginBottom "0.5rem"}}
+          [icons/ExceptionIcon {:size 18 :color text-color}]
+          [:span {:style {:marginLeft "0.5rem" :fontWeight "bold"
+                          :verticalAlign "middle"}}
+           (if title title "Service Alert")]]
+         [:div {:style {:color text-color :fontSize "90%"}}
+          (str message " ")
+          (if more-content
+            [:div {} [:a {:href  "javascript:;" :onClick #(swap! state assoc :showing-more? (not (:showing-more? @state)))
+                          :style {:color "#000"}} (if (:showing-more? @state) "Hide details..." "Show details...")]])
+          (when (:showing-more? @state) more-content)
+          (when link link)]])])})
+
+(react/defc BannerContainer
   {:get-initial-state
    (fn []
-     {:failedRetries 0})
+     {:failed-retries 0})
    :render
    (fn [{:keys [this state]}]
-     (when-let [alerts (not-empty (:alerts @state))]
-       [:div {}
+     (let [service-alerts (:service-alerts @state) js-alerts (:js-alerts @state)]
+       ; Why was zIndex of 514 chosen? IGV has a zIndex of 512, so modals use a zIndex of 513
+       ; We want these banners to be shown in front of the modals, so we use a zIndex of 514
+       [:div {:style {:zIndex 514 :position "relative"}}
+        (map (fn [alert]
```

Comment:
So, I'd actually prefer if `service-alerts` and `js-alerts` could just be two separate instances of `BannerContainer`, rather than making the actual content of the component a part of its display logic, as this prevents it from being used generically in the future.